blocks/procedures.js

```diff
@@ -1026,14 +1110,20 @@ Blockly.Blocks['procedures_ifreturn'] = {
           .appendField(Blockly.Msg['PROCEDURES_DEFRETURN_RETURN']);
     }
   },
+
+  // This block does not need JSO serialization hooks (saveExtraState and
+  // loadExtraState) because the state of this block is already encoded in the
+  // block's position in the workspace.
+  // XML hooks are kept for backwards compatibility.
+
   /**
    * Called whenever anything on the workspace changes.
    * Add warning if this flow block is not nested inside a loop.
    * @param {!Blockly.Events.Abstract} _e Change event.
    * @this {Blockly.Block}
    */
   onchange: function(_e) {
-    if (!this.workspace.isDragging || this.workspace.isDragging()) {
+    if (this.workspace.isDragging && this.workspace.isDragging()) {
```

Comment:
Are you sure this is intended? The old effect would basically short circuit early if we had a Workspace instead of a WorkspaceSvg. In WorkspaceSvg `isDragging` would never be undefined, it looks like to me. So this is changing the behavior to have this function run on non-svg workspaces and I don't know enough to know if it should or not.

Reply:
Yeah I changed it because I do want it to run on headless workspaces. Before this wasn't as much of a problem because we were serializing the state (ie whether the input exists or not). But because I didn't add jso serialization hooks, now this relies on the change listener to remove the input.

Plus this really should have been running in headless anyway, because if devs programmatically move their blocks we want the block to react how it would in a rendered workspace. Eg, this is important for automated testing.